packages/actor-query-operation-leftjoin/lib/ActorQueryOperationLeftJoin.ts

```diff
@@ -36,12 +36,19 @@ export class ActorQueryOperationLeftJoin extends ActorQueryOperationTypedMediate
 
     // If the pattern contains an expression, filter the resulting stream
     if (operationOriginal.expression) {
+      const rightMetadata = await entries[1].output.metadata();
+      const expressionVariables = rightMetadata.variables;
       const config = { ...ActorQueryOperation.getAsyncExpressionContext(context) };
       const evaluator = new AsyncEvaluator(operationOriginal.expression, config);
       const bindingsStream = joined.bindingsStream
         .transform({
           autoStart: false,
           transform: async(bindings: Bindings, done: () => void, push: (item: Bindings) => void) => {
+            if (!expressionVariables.every(variable => bindings.has(variable.value))) {
```

Comment:
Could you add the following comment before this line?:
`// If variables of the right-hand entry are missing, we skip expression evaluation`